src/routers/handlers/check_details/index.ts

```diff
@@ -33,12 +40,39 @@ export class CheckDetailsHandler extends GenericHandler<CheckDetailsViewData> {
             throw new Error("Presenter account address has not been set.");
         }
 
+        if (details.isBusinessRegistered === undefined) {
+            throw new Error("Presenter account is business registered has not been set.");
+        }
+
+        if (details.name === undefined || details.name.forename === null || details.name.surname === null) {
+            throw new Error("Presenter account name/forename/surname has not been set.");
+        }
+
+
+        // if isBusinessRegistered is true; company name need to be set
+        if (details.companyName === undefined && details.isBusinessRegistered) {
+            throw new Error("Presenter account company name has not been set for a registered business.");
+        }
+
+        // if isBusinessRegistered is false; business name need to be set
+        if (details.businessName === undefined && !details.isBusinessRegistered) {
+            throw new Error("Presenter account business name has not been set for a registered business.");
+        }
```

Comment:
Could we encapsulate this in it's own function, this is adding quite a lot of lines to the function